src/pyodide-py/pyodide/console.py

```diff
@@ -0,0 +1,153 @@
+        Wether or not the std redirection should be kept between calls to
+        `runcode`.
+    """
+
+    def __init__(
+        self,
+        locals: Optional[dict] = None,
+        stdout_callback: Optional[Callable[[str], None]] = None,
+        stderr_callback: Optional[Callable[[str], None]] = None,
+        persistent_stream_redirection: bool = False,
+    ):
+        super().__init__(locals)
+        self._stdout = None
+        self._stderr = None
+        self.stdout_callback = stdout_callback
+        self.stderr_callback = stderr_callback
+        self._persistent_stream_redirection = persistent_stream_redirection
+        if self._persistent_stream_redirection:
+            self.redirect_stdstreams()
+
+    def redirect_stdstreams(self):
+        """ Toggle stdout/stderr redirections. """
+
+        if self._stdout is None:
+            # we use meta callbacks to allow self.std{out,err}_callback
+            # overloading.
+            # we check callback against None at each call since it can be
+            # changed dynamically.
+            def meta_stdout_callback(*args):
+                if self.stdout_callback is not None:
+                    return self.stdout_callback(*args)
+
+            # for later restore
+            self._old_stdout = sys.stdout
+
+            # it would be more robust to use sys.stdout.name but testing
+            # system oveload them. Anyway it should be pretty stable
+            # upstream.
+            self._stdout = _StdStream(meta_stdout_callback, name="<stdout>")
+
+        if self._stderr is None:
+
+            def meta_stderr_callback(*args):
+                if self.stderr_callback is not None:
+                    return self.stderr_callback(*args)
+
+            self._old_stderr = sys.stderr
+            self._stderr = _StdStream(meta_stderr_callback, name="<stderr>")
+
+        # actual redirection
+        sys.stdout = self._stdout
+        sys.stderr = self._stderr
+
+    def restore_stdstreams(self):
+        """Restore stdout/stderr to the value it was before
+        the creation of the object."""
+        sys.stdout = self._old_stdout
+        sys.stderr = self._old_stderr
+
+    def runcode(self, code):
+        if self._persistent_stream_redirection:
+            super().runcode(code)
+        else:
+            self.redirect_stdstreams()
+            super().runcode(code)
+            self.restore_stdstreams()
+
+    def __del__(self):
+        if self._persistent_stream_redirection:
+            self.restore_stdstreams()
+
+    def banner(self):
+        """ A banner similar to the one printed by the real Python interpreter. """
+        # copyied from https://github.com/python/cpython/blob/799f8489d418b7f9207d333eac38214931bd7dcc/Lib/code.py#L214
+        cprt = 'Type "help", "copyright", "credits" or "license" for more information.'
+        return f"Python {platform.python_version()} {platform.python_build()} on WebAssembly VM\n{cprt}"
```

Comment:
This currently prints,
```
Python 3.8.2 ('default', 'Jan 15 2021 09:54:23') on WebAssembly VM
```
while the classical Python REPL displays,
```
Python 3.8.6 (default, Sep 25 2020, 09:36:53)
```
It would be nice to remove the quotes.

Reply:
Arf, sorry for this. Good catch! It will be corrected in #1141. Is that OK?

Reply:
Yes, sure, thanks!